### PR TITLE
Add event collector and derived tstamps

### PR DESCRIPTION
**Pull request by Omicron7**
Also includes other changes which are running in production, but not reflected in master yet.

### CODE REVIEW

**Comment by mattdrees** [comment]
@Omicron7 can I hard-reset staging to this, and deploy to stage? (maybe once the travis checks pass?)

File: api/index.js [hunk 1]
```diff
@@ -17,17 +17,9 @@ api.use(jwt({ secret: process.env.JWT_SECRET }).unless((request) => {
...
-
-    if (incomingOrigin && (
-        incomingOrigin.startsWith('http://content-scoring-local.cru.org:') ||
-        incomingOrigin === 'https://content-scoring-stage.cru.org')) {
-      response.header('Access-Control-Allow-Origin', incomingOrigin)
-      response.header('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept')
-    }
-  }
-
+  response.header('Access-Control-Allow-Origin', '*')
```
Comment (wrandall22) [inline api/index.js]: Will this open a security vulnerability to DoS attacks?

**Comment by Omicron7** [comment]
@mattdrees Yes, seems reasonable.

File: models/event.js [hunk 2]
```diff
@@ -218,7 +220,7 @@ Event.Fields = Fields
 // prototype Fields on Event instances
 forEach(Fields, (index, key) => {
   // Skip event_id and user_id, these are Event columns
```
Comment (wrandall22) [inline models/event.js]: Should probably update this comment

File: models/event.js [hunk 3]
```diff
@@ -245,10 +247,10 @@ Event.fromRecord = (record) => {
   }
   const event = new Event()
   event.event_id = decoded[Fields.event_id]
-  // Use collector timestamp as event creation time, forcing UTC
-  event.created_at = new Date(decoded[Fields.collector_tstamp] + '+0000')
```
Comment (wrandall22) [inline models/event.js]: What will go in `created_at` with this line removed?

File: api/index.js (hunk 1 above)
Comment (Omicron7) [reply api/index.js]: No, as any malicious request can ignore CORS and hit the API with or without this. If necessary, we could whitelist which domains we will allow, but I don't think it's necessary.

File: models/event.js (hunk 3 above)
Comment (Omicron7) [reply models/event.js]: By leaving it empty it will get set to the Date/Time the event is created in the database.

File: models/event.js (hunk 2 above)
Comment (Omicron7) [reply models/event.js]: done.

**Comment by mattdrees** [comment]
I pushed this to 'staging', and it deployed there ok. I think we can merge, and verify it deploys to prod ok.

**Pull request merged**